pyuvdata/uvdata/uvdata.py

```diff
@@ -5158,17 +5213,28 @@ def _select_preprocess(
             else:
                 blt_inds = ant_blt_inds
 
-        if times is not None:
-            if time_range is not None:
-                raise ValueError('Only one of "times" and "time_range" can be set')
+        have_times = times is not None
+        have_time_range = time_range is not None
+        have_lsts = lsts is not None
+        have_lst_range = lst_range is not None
+        if (
+            np.count_nonzero([have_times, have_time_range, have_lsts, have_lst_range])
+            > 1
+        ):
+            raise ValueError(
+                "Only one of [times, time_range, lsts, lst_range] may be "
+                "specified per selection operation."
+            )
 
+        if times is not None:
             times = uvutils._get_iterable(times)
             if np.array(times).ndim > 1:
                 times = np.array(times).flatten()
 
             time_blt_inds = np.zeros(0, dtype=np.int64)
             for jd in times:
                 if jd in self.time_array:
+                    # TODO: maybe use a more tolerant check
```

Comment:
Should we turn this into an issue?

Reply:
Thanks for catching this! I've changed it to use `np.isclose` to test for membership, rather than `in`.